C/Cling/build_tarballs.jl

```diff
@@ -0,0 +1,136 @@
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/*
+popd
+
+cp src/LICENSE.TXT LICENSE_LLVM.TXT 
+cp src/tools/clang/LICENSE.TXT LICENSE_CLANG.TXT 
+cp src/tools/cling/LICENSE.TXT LICENSE_CLING.TXT 
+install_license LICENSE_LLVM.TXT 
+install_license LICENSE_CLANG.TXT 
+install_license LICENSE_CLING.TXT 
+
+# The very first thing we need to do is to build llvm-tblgen for x86_64-linux-muslc
+# This is because LLVM's cross-compile setup is kind of borked, so we just
+# build the tools natively ourselves, directly.  :/
+# See: https://github.com/JuliaPackaging/Yggdrasil/blob/86a4221f71ee8ca5c0fd1056829a1fe2b28c879b/M/Metal_LLVM_Tools/build_tarballs.jl#L35
+
+# Build llvm-tblgen and llvm-config
+mkdir ${WORKSPACE}/bootstrap
+pushd ${WORKSPACE}/bootstrap
+CMAKE_FLAGS=()
+CMAKE_FLAGS+=(-DLLVM_HOST_TRIPLE=${MACHTYPE})
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+CMAKE_FLAGS+=(-DLLVM_TARGETS_TO_BUILD="host;NVPTX")
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_HOST_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=OFF)
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
+ninja -j${nproc} llvm-tblgen clang-tblgen llvm-config
+popd
+
+# Let's do the actual build within the `build` subdirectory
+mkdir ${WORKSPACE}/build
+cd ${WORKSPACE}/build
+
+CMAKE_FLAGS=()
+
+# Tell LLVM where our pre-built tblgen tools are
+CMAKE_FLAGS+=(-DLLVM_TABLEGEN=${WORKSPACE}/bootstrap/bin/llvm-tblgen)
+CMAKE_FLAGS+=(-DCLANG_TABLEGEN=${WORKSPACE}/bootstrap/bin/clang-tblgen)
+CMAKE_FLAGS+=(-DLLVM_CONFIG_PATH=${WORKSPACE}/bootstrap/bin/llvm-config)
+
+# Install things into $prefix
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+
+# Explicitly use our cmake toolchain file and tell CMake we're cross-compiling
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=ON)
+
+if [[ "${target}" == *apple* ]]; then
+    # Without this, cling will try to call `x86_64-apple-darwin14-clang++`
+    # on the macOS machine to find C++ header locations.
+    CMAKE_FLAGS+=(-DCLING_CXX_PATH=clang++)
+fi
+
+# Release build for best performance
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
+ninja -j${nproc} \
+    tools/cling/install \
+    install-clang-resource-headers
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms(;exclude=x->
+    startswith(x.tags["arch"], r"arm|power") ||
+    Sys.isfreebsd(x) ||
+    Sys.iswindows(x) ||
+    x == Platform("aarch64", "macos"))
+    # FreeBSD build failed with:
+    #   libc.so.7: undefined reference to `__progname', `environ`
+    #
+    # Windows build failed with:
+    #   TCHAR.H, Shlwapi.h: No such file or directory
+    #
+    # aarch64 macOS failed with:
+    #   The C compiler is not able to compile a simple test program.
+    #
+    # Failed build Logs:
+    # https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=23989&view=results
+
```

Comment:
From [the log](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=23991&view=logs&j=dbb69af8-b1ef-5989-5332-212aaaaafdc5&t=84d17446-8eed-5d1e-9b65-1a5d4f22fc69&l=7392):
```suggestion
platforms = expand_cxxstring_abis(platforms)

```